docs/design/coreclr/botr/guide-for-porting.md

```diff
@@ -0,0 +1,437 @@
+        address. Current implementations require that the return address always
+        be on the stack to do so, although this is a known performance
+        deficiency for RISC platforms on arm64.
+
+2.  Architecture specific relocation information (to represent generation of
+    relocations for use by load, store, jmp and call instructions) See
+    <https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#coff-relocations-object-only>
+    for the sort of details that need to be defined.
+
+3.  Behavior and accessibility of processor single step features from within a
+    process. On Unix the CLR debugger uses an in process thread to single step
+    through functions.
+
+4.  Unwind information. CoreCLR uses Windows style unwind data internally, even
+    on Unix platforms. A Windows style unwind structure must be defined. In
+    addition, it is possible to enable generation of DWARF data for exposure
+    through the GDB JIT
+    <https://sourceware.org/gdb/onlinedocs/gdb/JIT-Interface.html> . This
+    support is conditional on an \#ifdef, but has been used in the past to
+    support bring up of new platforms.
+
+5.  EH Funclets. .NET requires a 2 pass exception model in order to properly
+    support exception filters. This substantially differs from the typical
+    Itanium ABI model which is used on most Linux architectures
+
+6.  OS behavior with Signals. Especially exactly where the reported instruction
+    pointer is located.
+
+7.  Little vs big endian. While.NET runtimes have been ported to big endian in
+    the past (notable examples include Mono support various game consoles, and
+    POWER, and XNA support on Xbox360) there are no current ports of CoreCLR to
+    a big endian platform.
+
+Components of the Runtime affected by a port to a new architecture
+==================================================================
+
+Notable components
+
+1.  The JIT. The jit maintains the largest concentration of architecture
+    specific logic in the stack. This is not surprising. See [Porting RyuJit](porting-ryujit.md)
+    for guidance.
+
+2.  The CLR VM. The VM is a mix of completely architecture neutral logic, and
+    very machine specific paths.
+
+3.  System.Private.CoreLib/System.Reflection. There is little to no architecture
+    specific work here that is necessary for bringup. Nice-to-have work involves
+    adding support for the architecture in the
+    System.Reflection.ImageFileMachine enum, and the ProcessorArchitecture enum,
+    and logic that manipulates it.
+
+4.  PE File format changes to add a new architecture. Also, the C# compiler likely
+    also needs a new switch to generate machine specific code for the new 
+    architecture.
+
+5.  Crossgen/Crossgen2 - As the AOT compilers that produce machine specific logic
+    from general purpose MSIL, these will be needed to improve startup performance.
+
+6.  R2RDump - This allows diagnosing issues in pre-compiled code.
```

Comment:
coredistools - Necessary for GCStress (if determining instruction boundaries is non-trivial), as well as for SuperPMI asm diffs for JIT development.